### ISSUE TITLE
query for a reference does not work

### ISSUE THREAD

**Issue by grundid** [pulyaevskiy/firebase-admin-interop#12]
When I try to query a collection for a document with a reference value I get the following error:

> Error: Argument "value" is not a valid FieldValue. Detected an object of type "DocumentReference" that doesn't match the expected instance. Please ensure that the Firestore types you are using are from the same NPM package.

**Comment by pulyaevskiy** [comment]
Fixed and released as 1.0.0-dev.14.0.

**Comment by grundid** [comment]
Thanks for the update. The queries are working now.

However I have a similar problem, when I try to store a document with a property that is an array of references. Looking at the firestore online editor in the firebase console this is possible/allowed.

The error messages says:

> Error: Detected an object of type "DocumentReference" that doesn't match the expected instance. Please ensure that the Firestore types you are using are from the same NPM package.

**Comment by pulyaevskiy** [comment]
Right, complex values in lists are not supported currently. There is nothing that prevents us from making it work though.

I created #24 to track this issue.

**Comment by pulyaevskiy** [comment]
Just an FYI, a new version is up on Pub with support for complex types in Firestore lists.
